### PR TITLE
Add new rule no-ember-testing-in-module-scope

### PR DESCRIPTION
**Pull request by tmquinn**
First draft, please comment especially with more tests or edge cases, error messages and any suggestions for which configurations this should go in

### CODE REVIEW

File: docs/rules/no-ember-testing-in-module-scope.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+## Ember.testing is not allowed in modules scope
+
+### Rule name: `no-ember-testing-in-module-scope`
+
+Since ember-cli-qunit@4.1.0 / ember-qunit@3.0.0, Ember.testing is only set to
```
Comment (rwjblue) [inline docs/rules/no-ember-testing-in-module-scope.md]: Can you wrap `Ember.testing` in backticks (for better highlighting) throughout this file?

File: docs/rules/no-ember-testing-in-module-scope.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+true while a test is executing instead of all the time. Also, Ember.testing is a
+getter/setter in Ember and destructuring will only read its value at the time
+the module is evaluated.
```
Comment (rwjblue) [inline docs/rules/no-ember-testing-in-module-scope.md]: s/the module is evaluated/of destructuring/

File: docs/rules/no-ember-testing-in-module-scope.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+https://github.com/emberjs/ember-test-helpers/pull/227
+
+```javascript
+// Good
+export default Component.extend({
+  init() {
+    this.isTesting = Ember.testing
```
Comment (rwjblue) [inline docs/rules/no-ember-testing-in-module-scope.md]: I don't think this should be recommended either. `Ember.testing` should be used directly wherever you need it (instead of this example using `if (this.isTesting) {}` somewhere).

File: lib/rules/no-ember-testing-in-module-scope.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const messages = [
+  'Ember.testing is not set in module scope',
+  'Can not use destructuring to reference testing',
```
Comment (rwjblue) [inline lib/rules/no-ember-testing-in-module-scope.js]: s/testing/Ember.testing/

File: tests/lib/rules/no-ember-testing-in-module-scope.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    {
+      code: 'foo.testing = true;'
+    },
+    {
+      code: `
+        import Ember from 'ember';
+
+        export default Ember.Component.extend({
+          init() {
+            this.isTesting = Ember.testing;
```
Comment (rwjblue) [inline tests/lib/rules/no-ember-testing-in-module-scope.js]: IMHO we should also forbid assignment expressions where the right hand side is `Ember.testing`...

File: docs/rules/no-ember-testing-in-module-scope.md (hunk 3 above)
Comment (tmquinn) [reply docs/rules/no-ember-testing-in-module-scope.md]: Should I just remove it from the docs, or should i change the rule to disallow any assignment of `Ember.testing`?

File: tests/lib/rules/no-ember-testing-in-module-scope.js (hunk 5 above)
Comment (tmquinn) [reply tests/lib/rules/no-ember-testing-in-module-scope.js]: d'oh

File: docs/rules/no-ember-testing-in-module-scope.md (hunk 3 above)
Comment (tmquinn) [reply docs/rules/no-ember-testing-in-module-scope.md]: I see this quation has been answered below

File: docs/rules/no-ember-testing-in-module-scope.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+Since ember-cli-qunit@4.1.0 / ember-qunit@3.0.0, `Ember.testing` is only set to
+true while a test is executing instead of all the time. Also, `Ember.testing` is a
+getter/setter in Ember and destructuring will only read its value at the time
+of destructuring.
+https://github.com/emberjs/ember-test-helpers/pull/227
+
+```javascript
+// Good
+export default Component.extend({
```
Comment (tmquinn) [inline docs/rules/no-ember-testing-in-module-scope.md]: uggh forgot to change this
